### PR TITLE
Save system logs in functional tests if server crashed

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Need these logs to debug #36610

### CODE REVIEW

**Review by qoega — APPROVED** [review]

**Comment by tavplubix** [comment]
```
+ for table in query_log zookeeper_log trace_log transactions_info_log
+ clickhouse-local --path /var/lib/clickhouse/ -q 'select * from system.query_log format TSVWithNamesAndTypes'
+ [[ -n '' ]]
+ for table in query_log zookeeper_log trace_log transactions_info_log
+ pigz
+ clickhouse-local --path /var/lib/clickhouse/ -q 'select * from system.zookeeper_log format TSVWithNamesAndTypes'
+ [[ -n '' ]]
+ for table in query_log zookeeper_log trace_log transactions_info_log
+ pigz
+ clickhouse-local --path /var/lib/clickhouse/ -q 'select * from system.trace_log format TSVWithNamesAndTypes'
+ [[ -n '' ]]
+ for table in query_log zookeeper_log trace_log transactions_info_log
+ pigz
+ clickhouse-local --path /var/lib/clickhouse/ -q 'select * from system.transactions_info_log format TSVWithNamesAndTypes'
+ [[ -n '' ]]
+ wait
+ pigz
Code: 76. DB::Exception: Cannot lock file /var/lib/clickhouse/status. Another server instance in same directory is already running. (CANNOT_OPEN_FILE)
Code: 76. DB::Exception: Cannot lock file /var/lib/clickhouse/status. Another server instance in same directory is already running. (CANNOT_OPEN_FILE)
Code: 76. DB::Exception: Cannot lock file /var/lib/clickhouse/status. Another server instance in same directory is already running. (CANNOT_OPEN_FILE)
```

**Comment by tavplubix** [comment]
Now it looks ok
Stateless tests (release, DatabaseReplicated, actions) [2/2] - server failed to start:
```
2022.05.04 00:21:18.912276 [ 557 ] {} <Warning> RaftInstance: Election timeout, initiate leader election
2022.05.04 00:21:19.402138 [ 483 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.402151 [ 484 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.402764 [ 484 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.402984 [ 484 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.403298 [ 484 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.403502 [ 484 ] {} <Warning> KeeperTCPHandler: Ignoring user request, because the server is not active yet
2022.05.04 00:21:19.404005 [ 1166 ] {} <Error> virtual bool DB::DDLWorker::initializeMainThread(): Code: 999. Coordination::Exception: All connection tries failed while connecting to ZooKeeper. nodes: [::1]:19181, [::1]:9181, [::1]:29181
Code: 999. Coordination::Exception: Keeper server rejected the connection during the handshake. Possibly it's overloaded, doesn't see leader or stale (Connection loss): while receiving handshake from ZooKeeper. (KEEPER_EXCEPTION) (version 22.5.1.1), [::1]:19181
Poco::Except
```

Anyway, I forgot to do another change

File: docker/test/stateless/run.sh [hunk 1]
```diff
@@ -143,18 +160,17 @@ pigz < /var/log/clickhouse-server/clickhouse-server.log > /test_output/clickhous
 #   for files >64MB, we want this files to be compressed explicitly
 for table in query_log zookeeper_log trace_log transactions_info_log
 do
-    clickhouse-client -q "select * from system.$table format TSVWithNamesAndTypes" | pigz > /test_output/$table.tsv.gz &
```
Comment (azat) [inline docker/test/stateless/run.sh:146]: The problem with using `clickhouse-local` here is that now it is not capable to save data from S3 storage, like the comment above this lines states. I guess `clickhouse-local` can be taught to use S3 (maybe simply using server config will be enough, though not sure)

Comment (tavplubix) [reply docker/test/stateless/run.sh:146]: https://github.com/ClickHouse/ClickHouse/pull/54198
https://github.com/ClickHouse/ClickHouse/pull/54246

**Pull request merged**